internal/plugin/transport.go

```diff
@@ -70,7 +84,31 @@ func NewTransportHandle(name string, t envelope.Transport) (Handle, error) {
 	if handshake.APIVersion != api.APIVersion {
 		return nil, errHandshakeFailed{
 			Name:   name,
-			Reason: errVersionMismatch{Want: api.APIVersion, Got: handshake.APIVersion},
+			Reason: errAPIVersionMismatch{Want: api.APIVersion, Got: handshake.APIVersion},
```

Comment:
with our offline discussions, I get it now. Let's just rename .Version for better distinction.